Branch2

anshgrover21 opened the pull request:
all changes done

Commit:
first commit

harshkashyap234 committed:
initial commit

Commit:
initial commit

Commit:
remove nested looping and using ObjectMapper and Json Objecet to map Data

Commit:
remove nested loops, add exceptions and also use ObjectMapper instead of using Mapstruct

Commit:
changes done

Commit:
Merge branch 'generateExcelFile' into branch2

harshkashyap234 commented on MongoToExcel/src/main/java/com/example/MongoToExcel/Controller/CommonController.java:
```diff
@@ -0,0 +1,37 @@
+    public String createFile(@RequestParam("orgNo") String orgNo , @RequestParam("startDate") String startDate, @RequestParam("endDate") String endDate) throws Exception {
+
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern(YYYY_MM_DD);
+
```
remove unwanted spaces

harshkashyap234 commented on MongoToExcel/src/main/java/com/example/MongoToExcel/Controller/CommonController.java:
```diff
@@ -0,0 +1,37 @@
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
```
remove unwanted spaces

harshkashyap234 commented on MongoToExcel/src/main/java/com/example/MongoToExcel/Service/CommonService.java:
```diff
@@ -0,0 +1,157 @@
+        calendar.set(Calendar.MILLISECOND, 999);
+        endDateDate = calendar.getTime();
+        startDate = outputFormat.format(startDateDate);
+        endDate = outputFormat.format(endDateDate);
```
make a different function to change the format of start and end date

harshkashyap234 commented on MongoToExcel/src/main/java/com/example/MongoToExcel/Controller/CommonController.java:
```diff
@@ -0,0 +1,37 @@
+    @RequestMapping()
+    public String createFile(@RequestParam("orgNo") String orgNo , @RequestParam("startDate") String startDate, @RequestParam("endDate") String endDate) throws Exception {
+
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern(YYYY_MM_DD);
```
dont write logic inside controller, pass the params directly in service and then do the further steps

harshkashyap234 commented on MongoToExcel/src/main/java/com/example/MongoToExcel/Exception/CommonExceptionHandler.java:
```diff
@@ -0,0 +1,16 @@
+        public ResponseEntity<Object> handle(Exception e){
+            return new ResponseEntity<>(e.getMessage(), HttpStatus.BAD_REQUEST);
+        }
+
```
similarly remove spaces form here

harshkashyap234 commented on MongoToExcel/src/main/java/com/example/MongoToExcel/Model/CommonConfig.java:
```diff
@@ -0,0 +1,12 @@
+
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.springframework.context.annotation.Configuration;
```
Is this file needed ?

harshkashyap234 commented on MongoToExcel/src/main/java/com/example/MongoToExcel/Service/CommonService.java:
```diff
@@ -0,0 +1,157 @@
+
+        int colno = 0;
+        XSSFRow  row = sheet.createRow(0);
+        Map<String,String> map = flattenUtil(order);
```
give proper naming to map

harshkashyap234 commented on MongoToExcel/src/main/java/com/example/MongoToExcel/Service/CommonService.java:
```diff
@@ -0,0 +1,157 @@
+    private CommonRepo commonRepo;
+    public static final String YYYY_MM_DD = "yyyy-MM-dd";
+
+
```
place all static strings together and all injected fields together

harshkashyap234 commented on MongoToExcel/src/main/java/com/example/MongoToExcel/Model/Orders.java:
```diff
@@ -0,0 +1,117 @@
+        // Getters and Setters
+    }
+
+
```
remove spaces

Commit:
changes suggested resolved

Commit:
changes suggested resolved

Commit:
added static map and staticMapProperties class for selective retrieval of columns

Commit:
added static map and staticMapProperties class for selective retrieval of columns

Commit:
Created new Module name SheetByChargeType for Task. Creating sheet for difference charge type

Commit:
Created new Module name SheetByChargeType for Task. Creating sheet for difference charge type

Commit:
Created new Module name SheetByChargeType for Task. Creating sheet for difference charge type

Commit:
Created new Module name SheetByChargeType for Task. Creating sheet for difference charge type

Commit:
changes header name as per Given Sheet

Commit:
column containing % data handled and also controller for fetching byte[] for transection referance as parameter also added (strictly stick on assumption of having only one unique chargetype for every transaction id)

Commit:
column containing % data handled and also controller for fetching byte[] for transection referance as parameter also added (strictly stick on assumption of having only one unique chargetype for every transaction id)

Commit:
1. Added Maps for all the sheets inlcuding those which don't have any data write now, (there flatten part is entered as per assumption)

2. Remove the usage of Linked HashMap with HashMap(as using Linked Hash Map makes column filling dependent on order of flatten map)

3. added a new int field in columnproperty class which represent column number so we are declaring column number in static map itself

Commit:
1. Created JSON files for every sheet
2. config file is also Optimised

Commit:
1. Added new API to execute query on transactions and reflect data in table in pdf

Commit:
1. Added new service clasS ApiService hitting api for pdf through web client for unique org code
